core/src/main/java/org/apache/gravitino/catalog/ModelOperationDispatcher.java

```diff
@@ -236,6 +239,53 @@ public Model alterModel(NameIdentifier ident, ModelChange... changes)
                 alteredModel.properties()));
   }
 
+  /** {@inheritDoc} */
+  @Override
+  public ModelVersion alterModelVersion(
+      NameIdentifier ident, int version, ModelVersionChange... changes)
+      throws NoSuchModelVersionException, IllegalArgumentException {
+    validateAlterProperties(ident, HasPropertyMetadata::modelPropertiesMetadata, changes);
+    return executeAlterModelVersion(ident, f -> f.alterModelVersion(ident, version, changes));
+  }
+
+  /** {@inheritDoc} */
+  @Override
+  public ModelVersion alterModelVersion(
+      NameIdentifier ident, String alias, ModelVersionChange... changes)
+      throws NoSuchModelException, IllegalArgumentException {
+    validateAlterProperties(ident, HasPropertyMetadata::modelPropertiesMetadata, changes);
+    return executeAlterModelVersion(ident, f -> f.alterModelVersion(ident, alias, changes));
+  }
+
+  private ModelVersion executeAlterModelVersion(
+      NameIdentifier ident, ThrowableFunction<ModelCatalog, ModelVersion> fn) {
+
+    NameIdentifier catalogIdent = getCatalogIdentifier(ident);
+
+    ModelVersion alteredModelVersion =
+        TreeLockUtils.doWithTreeLock(
+            ident,
+            LockType.WRITE,
+            () ->
+                doWithCatalog(
+                    catalogIdent,
+                    c -> c.doWithModelOps(fn),
+                    NoSuchModelVersionException.class,
+                    IllegalArgumentException.class));
+
+    return internalUpdateModelVersion(catalogIdent, alteredModelVersion);
+  }
+
+  private ModelVersion internalUpdateModelVersion(
```

Comment:
I don't see the value of having this method, there's only one place that uses this, I think we don't need this method. Besides, the method name is a bit weird, the name doesn't reflect the implementation.